addons/Joystick/lua/weapons/gmod_tool/stools/wire_joystick.lua

```diff
@@ -1,344 +1,412 @@
-			draw.DrawText("Joystick Tool","Trebuchet36",4,0,Color(255,255,255,255),0)
-			
-			local uid = tostring(LocalPlayer():GetInfo("wire_joystick_uid"))
-			if uid:sub(1,3) ~= "jm_" then
-				uid = "jm_"..uid
-			end
-			
-			draw.DrawText("UID: "..uid,"Trebuchet20",4,36,Color(255,255,255,255),0)
-			draw.DrawText("Type: "..tostring(LocalPlayer():GetInfo("wire_joystick_analog") == "1" and "analog" or "digital"),"Trebuchet20",w-4,36,Color(255,255,255,255),2)
-			
-			if not jcon then
-				return
-			end
-			local reg = jcon.getRegisterByUID(uid)
-			if reg and reg.IsJoystickReg then
-				if reg:IsBound() then
-					local val = reg:GetValue()
-					if type(val) == "number" then
-						surface.SetDrawColor(255,0,0,255)
-						surface.DrawRect(0,h/2-16,w,32)
-						surface.SetDrawColor(0,255,0,255)
-						local disp = w*((val-reg.min)/(reg.max-reg.min))
-						surface.DrawRect(0,h/2-16,disp,32)
-						
-						local text = tonumber(val) or 0
-						local max = tonumber(LocalPlayer():GetInfo("wire_joystick_max")) or 0
-						local min = tonumber(LocalPlayer():GetInfo("wire_joystick_min")) or 0
-						text = text/255*(max-min)+min
-						draw.DrawText(math.Round(text),"Trebuchet50",w/2,h/2-20,Color(0,0,255,255),1)
-					elseif type(val) == "boolean" then
-						surface.SetDrawColor(255,0,0,255)
-						surface.DrawRect(0,h/2-16,w,32)
-						surface.SetDrawColor(0,255,0,255)
-						if val then
-							surface.DrawRect(0,h/2-16,w,32)
-						end
-						local max = tonumber(LocalPlayer():GetInfo("wire_joystick_max")) or 0
-						local min = tonumber(LocalPlayer():GetInfo("wire_joystick_min")) or 0
-						draw.DrawText(val and max or min,"Trebuchet50",w/2,h/2-20,Color(0,0,255,255),1)
-					end
-					draw.DrawText(reg:GetDeviceName() or "","Trebuchet20",4,h-20,Color(255,255,255,255),0)
-				else
-					surface.SetDrawColor(255,165,0,255)
-					surface.DrawRect(0,h/2-16,w,32)
-					draw.DrawText(uid.." unbound","Trebuchet50",w/2,h/2-20,Color(0,0,255,255),1)
-				end
-			else
-				surface.SetDrawColor(32,178,170,255)
-				surface.DrawRect(0,h/2-16,w,32)
-				draw.DrawText(uid.." inactive","Trebuchet50",w/2,h/2-20,Color(0,0,255,255),1)
-			end
-		end)
-		if not b then
-			ErrorNoHalt(e,"\n")
-		end
-	end
-end
\ No newline at end of file
+  surface.CreateFont("Trebuchet50", {size = 50, weight = 500, antialias = true, font = "trebuchet"})
+  surface.CreateFont("Trebuchet36", {size = 36, weight = 500, antialias = true, font = "trebuchet"})
+
+  local clWhite = Color( 255, 250, 255, 255 )
+  local clBlack = Color(   0,   0,   0, 255 )
+  local clBlue  = Color(   0,   0, 255, 255 )
+  local clCyan  = Color(   0, 255, 255, 255 )
+  local clMagen = Color( 255,   0, 255, 255 )
+  local clRed   = Color( 255,   0,   0, 255 )
+  local clYello = Color( 255, 255,   0, 255 )
+  local clGreen = Color(   0, 255,   0, 255 )
+  local clInBnd = Color( 255, 165,   0, 255 )
+  local clInAct = Color(  32, 178, 170, 255 )
+
+  function TOOL:DrawToolScreen(w,h)
```

Comment:
Improved readability of the tool screen. When drawing a single bind, more space can be utilized